src/app/core/services/scheduling.service.ts

```diff
@@ -281,9 +281,9 @@ export class SchedulingService {
     const repeatP = assessment.protocol.repeatProtocol
     const repeatQ = assessment.protocol.repeatQuestionnaire
 
-    let iterDate = new Date(this.refTimestamp)
+    let iterDate = this.setDateTimeToMidnight(new Date(this.enrolmentDate))
     const yearsMillis = DefaultScheduleYearCoverage * 60000 * 60 * 24 * 365
```

Comment:
Please use the time functions

Reply:
Do you mean the time calculation functions? It is in the other pull request that hasn't been merged yet. Should I merge them here?

Reply:
Okay we can merge in 0.5.10